bot/exts/filtering/_ui/ui.py

```diff
@@ -531,12 +532,91 @@ async def cancel(self, interaction: Interaction, button: discord.ui.Button) -> N
         await interaction.response.edit_message(content="🚫 Operation canceled.", view=None)
 
 
+class PhishConfirmationView(discord.ui.View):
+    """Confirmation buttons for whether the alert was for a phishing attempt."""
+
+    def __init__(
+        self, mod: Member, offender: User | Member | None, phishing_content: str, target_filter_list: FilterList
+    ):
+        super().__init__()
+        self.mod = mod
+        self.offender = offender
+        self.phishing_content = phishing_content
+        self.target_filter_list = target_filter_list
+
+    async def interaction_check(self, interaction: Interaction) -> bool:
+        """Only allow interactions from the command invoker."""
+        return interaction.user.id == self.mod.id
+
+    @discord.ui.button(label="Do it", style=discord.ButtonStyle.green)
+    async def confirm(self, interaction: Interaction, button: discord.ui.Button) -> None:
+        """Auto-ban the user and add the phishing content to the appropriate filter list as an auto-ban filter."""
+        await interaction.response.edit_message(view=None)
+
+        if self.offender:
+            compban_command = bot.instance.get_command("compban")
+            if not compban_command:
+                await interaction.followup.send(':warning: Could not find the command "compban".')
+            else:
+                ctx = FakeContext(interaction.message, interaction.channel, compban_command, author=self.mod)
+                await compban_command(ctx, self.offender)
+
+        compf_command = bot.instance.get_command("compfilter")
+        if not compf_command:
+            message = ':warning: Could not find the command "compfilter".'
+            await interaction.followup.send(message)
+        else:
+            ctx = FakeContext(interaction.message, interaction.channel, compf_command)
+            await compf_command(ctx, self.target_filter_list.name, self.phishing_content)
+
+
+    @discord.ui.button(label="Cancel")
+    async def cancel(self, interaction: Interaction, button: discord.ui.Button) -> None:
+        """Cancel the operation."""
+        new_message_content = f"~~{interaction.message.content}~~ 🚫 Operation canceled."
+        await interaction.response.edit_message(content=new_message_content, view=None)
+
+class PhishHandlingButton(discord.ui.Button):
+    """
+    A button that handles a phishing attempt.
+
+    When pressed, ask for confirmation.
+    If confirmed, comp-ban the offending user, and add the appropriate domain or invite as an auto-ban filter.
+    """
+
+    def __init__(self, offender: User | Member | None, phishing_content: str, target_filter_list: FilterList):
+        super().__init__(emoji="🎣")
+        self.offender = offender
+        self.phishing_content = phishing_content
+        self.target_filter_list = target_filter_list
+
+    @lock_arg("phishing", "interaction", lambda interaction: interaction.message.id)
```

Comment:
lambda 😢 
```suggestion
    @lock_arg("phishing", "interaction", attrgetter("message.id"))
```
Needs `from operator import attrgetter`

Reply:
lambda good